src/lib/metrics.js

```diff
@@ -0,0 +1,334 @@
+// Use of package.json for configuration
+import packageMeta from '../../package.json';
+
+import { makeLog } from './utils';
+import { selectors } from './store';
+import { colorsWithAlpha } from './constants';
+
+const log = makeLog('metrics');
+
+const GA_URL = 'https://ssl.google-analytics.com/collect';
+const GA_TRACKING_ID = packageMeta.config.GA_TRACKING_ID;
+const GA_APP_ID = packageMeta.extensionManifest.applications.gecko.id;
+const GA_APP_VERSION = packageMeta.version;
+
+let clientUUID = null;
+
+// Time after which to consider add-on install to have failed, because we have
+// no good way to tell if the install has been cancelled - only if the add-on
+// becomes available
+const INSTALL_FAILURE_DELAY = 10 * 1000;
+let installTimer = null;
+
+// See also: https://github.com/mozilla/ThemesRFun/blob/master/docs/metrics.md
+
+// Custom metrics
```

Comment:
Some of these comments for dimensions are messy, because I copypasta'd from metrics.md and turned them into JS with a regex. Might clean them up, but figured there might be changes to metrics.md yet (#11)